File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.go
func cantorPair(y int, x int) int {
  return (y + x + 1) * (y + x) / 2 + y
}

func countServers(grid [][]int) int {
  h := len(grid)
  w := len(grid[0])
  rows := make([]map[int]bool, h)
  cols := make([]map[int]bool, w)
  for y := range rows {
    rows[y] = map[int]bool{}
  }
  for x := range cols {
    cols[x] = map[int]bool{}
  }
  for y, row := range grid {
    for x, val := range row {
      if val == 1 {
        pair := cantorPair(y, x);
        rows[y][pair] = true
        cols[x][pair] = true
      }
    }
  }
  ans := map[int]bool{}
  for _, row := range rows {
    if len(row) > 1 {
      maps.Insert(ans, maps.All(row))
    }
  }
  for _, col := range cols {
    if len(col) > 1 {
      maps.Insert(ans, maps.All(col))
    }
  }
  return len(ans)
}

File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.py
class Solution:
  def countServers(self, grid: List[List[int]]) -> int:
    rows = [set() for _ in range(len(grid))]
    cols = [set() for _ in range(len(grid[0]))]
    for y in range(len(grid)):
      for x in range(len(grid[0])):
        if grid[y][x] == 1:
          rows[y].add((y, x))
          cols[x].add((y, x))
    s = set()
    for row in rows:
      if len(row) > 1:
        s.update(row)
    for col in cols:
      if len(col) > 1:
        s.update(col)
    return len(s)

        